Automattic/wp-calypso
# RT: Use CPT page and mShots for previewing RTs

Fixes https://github.com/Automattic/dotcom-forge/issues/8617
Requires D157805-code

## Proposed Changes

- Moves away from the block renderer endpoint to generate RT previews because we're not able to simulate the look&feel of the frontend in there
- Instead we now preview the frontend of the RT pages and mShot them


Before | After
--- | ---
<img width="407" alt="Screenshot 2024-08-08 at 11 59 18" src="https://github.com/user-attachments/assets/5cba8285-1662-4ce3-a038-c0e58701ad3e"> | <img width="410" alt="Screenshot 2024-08-08 at 13 33 50" src="https://github.com/user-attachments/assets/073d614f-86e5-49de-972d-b29efe1bdd66">


- Note: I initially used the `WebPreview` component but had to abandon it because [Chrome has subpixel rendering issues when scaling the content with CSS](https://issues.chromium.org/issues/323982813). It's easier to just use mShots because there are no rendering issues when scaling down images.
- Note 2: Because we have slugs now (needed for the previews), I thought it'd be interesting to use them in the URLs instead of the IDs.

## Why are these changes being made?

Because the current RT previews don't reflect the actual layout applied to the site

## Testing Instructions

- Use the Calypso live link below
- Go to `/patterns`
- Scroll down to the RT section
- Make sure the preview of Cafe looks good now (no horizontal line between header and content, width of the drinks menu is wider)
- Open the RT details
- Make sure the preview still looks good

## Pre-merge Checklist
